pkg/parse/namespace.go

```diff
@@ -158,23 +158,38 @@ func (p *namespace) setSourceStatusWithRetries(ctx context.Context, newStatus so
 		return status.APIServerError(err, "failed to get RepoSync for parser")
 	}
 
+	currentRS := rs.DeepCopy()
+
 	setSourceStatus(&rs.Status.Source, p, newStatus, denominator)
 
-	continueSyncing := true
-	if rs.Status.Source.ErrorSummary.TotalCount > 0 {
-		continueSyncing = false
-	}
-	metrics.RecordPipelineError(ctx, configsync.RepoSyncName, "source", rs.Status.Source.ErrorSummary.TotalCount)
+	continueSyncing := (rs.Status.Source.ErrorSummary.TotalCount == 0)
 	var errorSource []v1beta1.ErrorSource
-	if rs.Status.Source.Errors != nil {
+	if len(rs.Status.Source.Errors) > 0 {
 		errorSource = []v1beta1.ErrorSource{v1beta1.SourceError}
 	}
 	reposync.SetSyncing(&rs, continueSyncing, "Source", "Source", newStatus.commit, errorSource, rs.Status.Source.ErrorSummary, newStatus.lastUpdate)
 
-	metrics.RecordReconcilerErrors(ctx, "source", status.ToCSE(newStatus.errs))
+	// Avoid unnecessary status updates.
+	if cmp.Equal(currentRS.Status, rs.Status, compare.IgnoreTimestampUpdates) {
+		klog.V(5).Infof("Skipping source status update for RepoSync %s/%s", rs.Namespace, rs.Name)
+		return nil
+	}
+
+	csErrs := status.ToCSE(newStatus.errs)
+	metrics.RecordReconcilerErrors(ctx, "source", csErrs)
+	metrics.RecordPipelineError(ctx, configsync.RepoSyncName, "source", len(csErrs))
```

Comment:
Seems we should define some constants for the component names (`source`, `rendering`, `sync`, etc)

Reply:
Aye. Can do in a follow up.